Usability improvements to build steps

antoineco opened the pull request:
**What this PR does / why we need it**:

Suggestions of tiny improvements for things I run often. I though some may make sense for the rest of us as well:

* Make the compiler actually use the [Go cache](https://golang.org/cmd/go/#hdr-Build_and_test_caching) (Go [1.10+](https://golang.org/doc/go1.10)) by avoiding full rebuilds (remove `-a` flag). Speeds up builds quite a lot on modest hardware.
* Make build flags parameterizable: `GOBUILD_FLAGS` variable, defaults to `-v` to show the build progress. Also makes the wait more meaningful than a long blank on modest hardware.
* Print main build steps ("+ Building bin/amd64/...", "+ Copying to temp dir...")
* ~~Move build of `ingress-nginx:build` image to a `build-image` make target, mainly to be able to log that step in the right order.~~
* Rename make `clean` target to `clean-container`, because it's really what it does.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:

Does not introduce any feature or fix any bug.

aledbf commented:
@antoineco we removed the local build in https://github.com/kubernetes/ingress-nginx/pull/2868
Besides that, this LGTM

aledbf commented on build/build.sh:
```diff
@@ -41,7 +41,8 @@ fi
 
-go build -a -installsuffix cgo 	\
+go build \
+    ${GOBUILD_FLAGS} \
```
you need to set a default in case this env value is not set (that's why this is failing CI)

antoineco commented:
@aledbf I addressed your comments and added some more minor fixes/improvements. Each commit should be self-documented.

Putting this on hold for now because I want to squash these commits.
/hold

antoineco commented on Makefile:
```diff
@@ -48,7 +48,7 @@ ARCH ?= $(shell go env GOARCH)
 DUMB_ARCH = ${ARCH}
 
-GOBUILD_FLAGS ?= -v
+GOBUILD_FLAGS :=
```
Tested with both `make build` and `make build GOBUILD_FLAGS='-v -a'`.
Does what's expected in both scenarios using a pretty old `make` version (3.81, mac OS).

antoineco commented on Makefile:
```diff
@@ -161,7 +161,7 @@ endif
 	@echo "+ Building bin/$(ARCH)/nginx-ingress-controller"
 	@$(DEF_VARS) \
-	GOBUILD_FLAGS=$(GOBUILD_FLAGS) \
+	GOBUILD_FLAGS="$(GOBUILD_FLAGS)" \
```
Quotes needed to avoid `/bin/sh: -a: command not found` when passing custom flags.

antoineco commented on Makefile:
```diff
@@ -164,9 +168,6 @@ build:
 	GOBUILD_FLAGS="$(GOBUILD_FLAGS)" \
 	build/go-in-docker.sh build/build.sh
 
-	@echo "+ Copying artifact to temporary directory"
```
I figured this didn't make sense here, in isolation, because `make build` and `make container` could not be invoked separately.

antoineco commented:
Squashed (I hope the thumb up reaction meant 'ok' 😉).
/hold cancel

aledbf commented:
/lgtm
